Failing X509_VERIFY_PARAM_set_hostflags with openssl-1.0.2-b2

```
x86_64-pc-linux-gnu-gcc -march=barcelona -mtune=barcelona -O2 -pipe -fPIC -I/usr/include/python3.3 -c cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_4f32d02dx3a8a382.c -o /var/tmp/portage/dev-python/cryptography-0.5.3/work/cryptography-0.5.3/cryptography/hazmat/bindings/__pycache__/cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_4f32d02dx3a8a382.o
cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_4f32d02dx3a8a382.c:2045:8: error: ‘X509_VERIFY_PARAM_set_hostflags’ redeclared as different kind of symbol
 void (*X509_VERIFY_PARAM_set_hostflags)(X509_VERIFY_PARAM *,
        ^
In file included from /usr/include/openssl/x509.h:601:0,
                 from /usr/include/openssl/cms.h:58,
                 from cryptography/hazmat/bindings/__pycache__/_Cryptography_cffi_4f32d02dx3a8a382.c:217:
/usr/include/openssl/x509_vfy.h:565:6: note: previous declaration of ‘X509_VERIFY_PARAM_set_hostflags’ was here
 void X509_VERIFY_PARAM_set_hostflags(X509_VERIFY_PARAM *param,
      ^
Traceback (most recent call last):
  File "/usr/lib64/python3.3/distutils/unixccompiler.py", line 126, in _compile
    extra_postargs)
  File "/usr/lib64/python3.3/distutils/ccompiler.py", line 909, in spawn
    spawn(cmd, dry_run=self.dry_run)
  File "/usr/lib64/python3.3/distutils/spawn.py", line 32, in spawn
    _spawn_posix(cmd, search_path, dry_run=dry_run)
  File "/usr/lib64/python3.3/distutils/spawn.py", line 163, in _spawn_posix
    % (cmd[0], exit_status))
distutils.errors.DistutilsExecError: command 'x86_64-pc-linux-gnu-gcc' failed with exit status 1
```

As reported here: https://bugs.gentoo.org/show_bug.cgi?id=519508.

--- a/cryptography/hazmat/bindings/openssl/x509_vfy.py
+++ b/cryptography/hazmat/bindings/openssl/x509_vfy.py
@@ -185,10 +185,10 @@ void X509_STORE_CTX_set0_crls(X509_STORE_CTX *,
                               Cryptography_STACK_OF_X509_CRL *);
 
 /* X509_VERIFY_PARAM */
-int X509_VERIFY_PARAM_set1_host(X509_VERIFY_PARAM *, const unsigned char *,
+int X509_VERIFY_PARAM_set1_host(X509_VERIFY_PARAM *, const char *,
                                 size_t);
 void X509_VERIFY_PARAM_set_hostflags(X509_VERIFY_PARAM *, unsigned int);
-int X509_VERIFY_PARAM_set1_email(X509_VERIFY_PARAM *, const unsigned char *,
+int X509_VERIFY_PARAM_set1_email(X509_VERIFY_PARAM *, const char *,
                                  size_t);
 int X509_VERIFY_PARAM_set1_ip(X509_VERIFY_PARAM *, const unsigned char *,
                               size_t);
@@ -224,9 +224,9 @@ static const long X509_V_FLAG_SUITEB_192_LOS = 0;
 static const long X509_V_FLAG_SUITEB_128_LOS = 0;
 static const long X509_V_FLAG_PARTIAL_CHAIN = 0;
 
-int (*X509_VERIFY_PARAM_set1_host)(X509_VERIFY_PARAM *, const unsigned char *,
+int (*X509_VERIFY_PARAM_set1_host)(X509_VERIFY_PARAM *, const char *,
                                    size_t) = NULL;
-int (*X509_VERIFY_PARAM_set1_email)(X509_VERIFY_PARAM *, const unsigned char *,
+int (*X509_VERIFY_PARAM_set1_email)(X509_VERIFY_PARAM *, const char *,
                                     size_t) = NULL;
 int (*X509_VERIFY_PARAM_set1_ip)(X509_VERIFY_PARAM *, const unsigned char *,
                                  size_t) = NULL;

